### PR TITLE
10/10 Fix to the ugly buttons

### PR DESCRIPTION
**Pull request by shedaniel**
Introduce yourself to buttons! These revolutionary widgets can do a wide variety of things when you interact with it, in this case, it will cycle through the pages in the creative menu, allowing more item groups that you can play with! Additionally, they also turn blue when you hover them! Wait a minute, what is this??
Bad picture
There is only one thing you could say to this situation: 

The buttons in the fabric creative menu are also flat and plastic: 
ugly
no not close enough, THIS:
uglybig
See? the background is flat and the text does not have a shadow, banned.

**With patch 1893123**, you can finally enjoy browsing through the creative menu!
Good picture
no not close enough, THIS:
good

10/10 fix!
<details>
<summary>Actual Changelogs for nerds</summary>
<p>
<ul>
<li>Fixed the button background so it is not flat</li>
<li>Added shadows to text</li>
<li>Moved & Resized the buttons a little bit so it looks better</li>
</ul>
</p>
</details>

### CODE REVIEW

File: fabric-item-groups-v0/src/main/java/net/fabricmc/fabric/impl/itemgroup/FabricCreativeGuiComponents.java [hunk 1]
```diff
@@ -47,24 +48,34 @@
...
 			if (this.visible) {
 				MinecraftClient minecraftClient = MinecraftClient.getInstance();
+				TextRenderer textRenderer = minecraftClient.textRenderer;
 				minecraftClient.getTextureManager().bindTexture(BUTTON_TEX);
 				GlStateManager.disableLighting();
 				GlStateManager.color4f(1F, 1F, 1F, 1F);
-				this.blit(this.x, this.y, (type == Type.NEXT ? 12 : 0), (active ? 0 : 12), 12, 12);
-
+				this.blit(this.x, this.y, 0, (active ? (isHovered() ? 24 : 0) : 12), 11, 11);
+				int textColor = 14737632;
```
Comment (gegy1000) [inline fabric-item-groups-v0/src/main/java/net/fabricmc/fabric/impl/itemgroup/FabricCreativeGuiComponents.java]: Could these be converted to hex?

Comment (shedaniel) [reply fabric-item-groups-v0/src/main/java/net/fabricmc/fabric/impl/itemgroup/FabricCreativeGuiComponents.java]: This is copied from vanilla but this is doable through manual conversion

**Comment by modmuss50** [comment]
Seems like a good improvement, better than my quick textures that I made.

File: fabric-item-groups-v0/src/main/java/net/fabricmc/fabric/impl/itemgroup/FabricCreativeGuiComponents.java (hunk 1 above)
Comment (Prospector) [reply fabric-item-groups-v0/src/main/java/net/fabricmc/fabric/impl/itemgroup/FabricCreativeGuiComponents.java]: image

**Comment by Prospector** [comment]
I still think we need to find another solution for the buttons as they're too small and still don't look great, but this is better than before

File: fabric-item-groups-v0/src/main/java/net/fabricmc/fabric/impl/itemgroup/FabricCreativeGuiComponents.java (hunk 1 above)
Comment (shedaniel) [reply fabric-item-groups-v0/src/main/java/net/fabricmc/fabric/impl/itemgroup/FabricCreativeGuiComponents.java]: why did i not know

**Comment by modmuss50** [comment]
Looks good, im going to move this onto Last Call.

**Comment by grondag** [comment]
Needs an update to minor version, otherwise looks good.

**Comment by shedaniel** [comment]
Tbh I'd say to wait for #323 or #316

**Comment by ExtraCrafTX** [comment]
I was quite fed up with the buttons myself and came up with a couple of my own before coming across this thread. About these ones, they were already quite small, making them smaller seems like a bad idea, especially when it no longer lines up with the search bar.

My ideas:
In the current position:
current-position
In between the tabs:
between-tabs
As tabs themselves (could also go in the bottom gap):
as-tab

The buttons could also go vertically on either side of the menu, as tabs or as buttons. They might look better as tabs on the side since there is a lot more space for them, they'd be able to be bigger as well.

I personally like the flat background as it matches with the rest of the creative menu. I'm undecided about the shadow, I think it crowds up the small amount of space unnecessarily.

**Comment by Prospector** [comment]
I like the small tab buttons in your last pic @ExtraCrafTX

**Comment by shedaniel** [comment]
I agree, my buttons are just a minor fix because they don’t look like buttons in vanilla. But in the last picture, that looks really really nice. I could make another PR for that if @ExtraCrafTX give me the textures

**Comment by ExtraCrafTX** [comment]
I already implemented them so I could either create a new pull request or I could send you the texture and you could do it, I don't mind either way.
I was thinking about whether the tab ones should highlight when you hover over them. Seeing as they look like tabs it might be weird if they highlighted while the other tabs don't, but at the same time, they are buttons...

**Comment by Prospector** [comment]
IMO as tabs they probably shouldn't.

**Comment by Sollace** [comment]
I don't think so either, as tabs. But should they move up/down when you hover over them?

**Comment by Runemoro** [comment]
The rounded buttons look really nice, but I think they should stay buttons, since they don't act as tabs at all (clicking them doesn't make them "selected" until you click another tab, they're just buttons that change which tabs are displayed).

**Comment by shedaniel** [comment]
IMO They should still turn blue when you hover them

**Comment by ExtraCrafTX** [comment]
Just for comparison,
Without highlighting:
without-highlight
With highlighting:
with-highlight

If we do keep them as buttons then they should definitely be highlighted.
If we make them look like tabs, IMO they still need some sort of feedback, whether that be highlighting or otherwise.
> I don't think so either, as tabs. But should they move up/down when you hover over them?

I can try out moving up and down, but since the two are kind of linked into one tab, it would be weird if only one moved.

**Comment by shedaniel** [comment]
@ExtraCrafTX Make other PR for that I guess. I will be closing this because yours look better.

**Comment by Sollace** [comment]
> I can try out moving up and down, but since the two are kind of linked into one tab, it would be weird if only one moved.
I don't think so, really. The buttons to me look distinct enough that they can be believed to be two tabs stuck together. When you hover over them, they would just move up 2-3 pixels to indicate that they're focused.

I don't exactly remember where, but I distinctly remember having that behaviour for creative screen's tabs at one point and it was quite nice.

**Comment by Runemoro** [comment]
The second version looks much better.

A tab is something that can be switched to by clicking on it. Clicking on something that looks like a tab but just acts like a button breaks users' expectations and makes the UI unintuitive and inconsistent.

**Comment by Prospector** [comment]
Is that highlighting accurate to how like the recipe book button looks? If so, 👍, if not maybe match it. (I don't actually remember how the recipe book one looks)

**Comment by ExtraCrafTX** [comment]
Yeah, the highlighting matches the recipe book button 👍

**Comment by Prospector** [comment]
yeah go ahead and open a PR then

**Comment by shedaniel** [comment]
Let me close this because let’s be real here #336 is better

**Pull request closed without merging**